packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,513 @@
+/// options. When one option in the segmented control is selected, the other
+/// options in the segmented control cease to be selected.
+///
+/// A segmented control can feature any [Widget] as one of the values in its
+/// [Map] of [children]. The type T is the type of the keys used
+/// to identify each widget and determine which widget is selected. As
+/// required by the [Map] class, keys must be of consistent types
+/// and must be comparable. The ordering of the keys will determine the order
+/// of the widgets in the segmented control.
+///
+/// When the state of the segmented control changes, the widget calls the
+/// [onValueChanged] callback. The key associated with the newly selected
+/// widget is returned in the [onValueChanged] callback. Most widgets that
+/// use a segmented control will listen for the [onValueChanged] callback
+/// and rebuild the segmented control with a new [groupValue] to update which
+/// option is currently selected.
+///
+/// The [children] will be displayed in the order of the keys in the [Map].
+/// The height of the segmented control is determined by the height of the
+/// tallest widget provided as a value in the [Map] of [children].
+/// The width of the segmented control is determined by the horizontal
+/// constraints on its parent. The available horizontal space is divided by
+/// the number of provided [children] to determine the width of each widget.
+/// The selection area for each of the widgets in the [Map] of
+/// [children] will then be expanded to fill the calculated space, so each
+/// widget will appear to have the same dimensions.
+///
+/// See also:
+///
+///  * <https://developer.apple.com/design/human-interface-guidelines/ios/controls/segmented-controls/>
+class SegmentedControl<T> extends StatefulWidget {
+  /// Creates an iOS-style segmented control bar.
+  ///
+  /// The [children] and [onValueChanged] arguments must not be null. The
+  /// [children] argument must be an ordered [Map] such as a [LinkedHashMap].
+  /// Further, the length of the [children] list must be greater than or
+  /// equal to two.
+  ///
+  /// Each widget value in the map of [children] must have an associated key
+  /// that uniquely identifies this widget. This key is what will be returned
+  /// in the [onValueChanged] callback when a new value from the [children] map
+  /// is selected.
+  ///
+  /// The [groupValue] must be one of the keys in the [children] map.
+  /// The [groupValue] is the currently selected value for the segmented control.
+  /// If no [groupValue] is provided, or the [groupValue] is null, no widget will
+  /// appear as selected.
+  SegmentedControl({
+    Key key,
+    @required this.children,
+    @required this.onValueChanged,
+    this.groupValue,
+  })  : assert(children != null),
+        assert(children.length >= 2),
+        assert(onValueChanged != null),
+        assert(groupValue == null || children.keys.any((T child) => child == groupValue)),
```

Comment:
You might add an additional argument to this assert that is a string saying something like "The group value must be one of the keys in the children map."  That string will be printed along with the assert if it fires, and makes it easier to read.

Reply:
It seems like this was not addressed.